fix : migrate root component to react functional component

divyanshu-patil opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->

## Proposed changes
Fix : #6031
this pull request migrate the root component from the React class component to the React functional component, ensuring that the code follows best react practices

## Issue(s)	
#6031 

## changed files
this going to change the root component in app/index.tsx

## Types of changes
<!-- What types of changes does your code introduce to Rocket.Chat? -->
<!-- Put an `x` in the boxes that apply -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative?pullRequest=6033) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative?pullRequest=6033) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/Rocket.Chat.ReactNative?pullRequest=6033) it.</sub>

Rohit3523 commented:
Hello, can you remove 'fixed #6031' from the title and start the title with 'fix: ' instead of '[FIX]' to align it with the PR title convention we follow.

Thanks

divyanshu-patil commented:
> Hello, can you remove 'fixed #6031' from the title and start the title with 'fix: ' instead of '[FIX]' to align it with the PR title convention we follow.
> 
> Thanks

done, thanks for feedback

divyanshu-patil commented:
> ## Proposed changes
> Fix : #6031 this pull request migrate the root component from the React class component to the React functional component, ensuring that the code follows best react practices
> 
> ## Issue(s)
> #6031
> 
> ## changed files
> this going to change the root component in app/index.tsx
> 
> ## Types of changes
> * [ ]  Bugfix (non-breaking change which fixes an issue)
> * [x]  Improvement (non-breaking change which improves a current function)
> * [ ]  New feature (non-breaking change which adds functionality)
> * [ ]  Documentation update (if none of the other choices apply)
> 
> ## Checklist
> * [x]  I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
> * [x]  I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
> * [x]  Lint and unit tests pass locally with my changes
> * [ ]  I have added tests that prove my fix is effective or that my feature works (if applicable)
> * [x]  I have added necessary documentation (if applicable)
> * [ ]  Any dependent changes have been merged and published in downstream modules
> 
> ## Further comments

diegolmello commented on app/index.tsx:
```diff
@@ -78,163 +78,164 @@ const parseDeepLinking = (url: string) => {
-	onDimensionsChange = debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
-		this.setDimensions({
+	// Handle dimensions change
+	const onDimensionsChange = debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
```
This kind of debounce won't work on function components.
You should use useDebounce or useDebounceHook

diegolmello commented on app/index.tsx:
```diff
@@ -78,163 +78,164 @@ const parseDeepLinking = (url: string) => {
-	// Dimensions update fires twice
-	onDimensionsChange = debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
-		this.setDimensions({
+	// Handle dimensions change
```
Is the old comment still valid?
No sense to change it otherwise.

divyanshu-patil replied:
I am learning about Rocket.chat, i will fix it

divyanshu-patil replied:
i have a question 
i gone through implementation of debounce but can you specify how debounce is going to called twice as immediate parameter is not provided, it should only execute once after the timeout function is called

divyanshu-patil replied:
> i have a question i gone through implementation of debounce but can you specify how `onDimensionsChange()` is going to called twice as immediate parameter is not provided, it should only execute once after the timeout function is called

nvm i misunderstood the comment

diegolmello commented on app/index.tsx:
```diff
@@ -78,163 +78,164 @@ const parseDeepLinking = (url: string) => {
+	});
+
+	useEffect(() => {
+		// init function
```
Remove unnecessary comments like this one.
ChatGPT?

divyanshu-patil replied:
nope i added it on my own 
while moving i used 2 files while putting all relevent code in the second file i added some comments
yes i used chatGPT but for understanding whats going on not copied the code from it
